File: src/main/java/gg/mineral/practice/managers/ArenaManager.java
package gg.mineral.practice.managers;

import gg.mineral.api.config.FileConfiguration;
import gg.mineral.practice.arena.Arena;
import it.unimi.dsi.fastutil.bytes.Byte2ObjectOpenHashMap;
import lombok.Getter;
import lombok.val;

public class ArenaManager {
	@Getter
	final static FileConfiguration config = new FileConfiguration("arenas.yml", "plugins/Practice");
	@Getter
	final static Byte2ObjectOpenHashMap<Arena> arenas = new Byte2ObjectOpenHashMap<>();
	public static byte CURRENT_ID = 0;

	public static void registerArena(Arena arena) {
		arenas.put(arena.getId(), arena);
	}

	public static void remove(Arena arena) {
		arenas.remove(arena.getId());
		arena.delete();

		for (val gametype : GametypeManager.getGametypes().values())
			gametype.getArenas().remove(arena.getId());

		for (val queuetype : QueuetypeManager.getQueuetypes().values())
			queuetype.getArenas().remove(arena.getId());
	}

	public static Arena getArenaByName(String string) {
		for (val arena : arenas.values())
			if (arena.getName().equalsIgnoreCase(string))
				return arena;

		return null;
	}

	public static void save() {
		for (val arena : getArenas().values())
			arena.save();

		config.save();
	}

	public static void load() {
		val configSection = getConfig().getConfigurationSection("Arena.");

		if (configSection == null) {
			setDefaults();
			return;
		}

		for (val key : configSection.getKeys(false)) {

			if (key == null)
				continue;

			val arena = new Arena(key, CURRENT_ID++);

			arena.load();

			registerArena(arena);
		}
	}

	public static void setDefaults() {
		val arena = new Arena("Default", CURRENT_ID++);
		arena.setDefaults();
		registerArena(arena);
	}
}

File: src/main/kotlin/gg/mineral/practice/managers/ArenaManager.kt
package gg.mineral.practice.managers

import gg.mineral.practice.arena.Arena
import gg.mineral.practice.arena.ArenaImpl
import gg.mineral.practice.util.config.yaml.FileConfiguration
import it.unimi.dsi.fastutil.bytes.Byte2ObjectOpenHashMap
import org.bukkit.configuration.ConfigurationSection

object ArenaManager {
    val config = FileConfiguration("arenas.yml", "plugins/Practice")
    val arenas: Byte2ObjectOpenHashMap<Arena?> = Byte2ObjectOpenHashMap()
    var CURRENT_ID: Byte = 0

    fun registerArena(arena: Arena): Arena? = arenas.put(arena.id, arena)

    fun remove(arena: Arena): Arena? {
        arena.delete()

        for (gametype in GametypeManager.gametypes.values) gametype?.arenas?.remove(arena.id)
        for (queuetype in QueuetypeManager.queuetypes.values) queuetype?.arenas?.remove(arena.id)
        return arenas.remove(arena.id)
    }

    fun getArenaByName(string: String): Arena? {
        for (arena in arenas.values) if (arena?.name.equals(string, ignoreCase = true)) return arena
        return null
    }

    fun load() {
        val configSection: ConfigurationSection = config.getConfigurationSection("Arena.") ?: return

        for (key in configSection.getKeys(false)) {
            if (key == null) continue
            registerArena(ArenaImpl(key, CURRENT_ID++))
        }
    }
}
